src/routes/+page.svelte

```diff
@@ -46,43 +47,54 @@
 	// State for sidebar menu show/hide toggle
 	let sidebarVisable = true;
 
-	// Load saved selected nodes from localStorage on component initialization
+	// State for filters
+	let highlightKeystones = false;
+	let highlightNotables = false;
+	let highlightSmalls = false;
+	let hideUnidentified = false;
+	let hideUnselected = false;
+	let hideSmall = false;
+
+	// Reactive statement for search
+	$: handleSearch(searchTerm);
+
 	if (browser) {
-		selectedAscendancy = localStorage.getItem('selectedAscendancy') || 'gemling';
+		const params = new URLSearchParams(window.location.search);
+		const asc = params.get('a');
+		const passivesCompressed = params.get('p');
 
-		const savedSelectedNodes = localStorage.getItem('selectedSkillNodes');
+		if (asc) {
+			selectedAscendancy = asc;
+		}
 
-		if (savedSelectedNodes) {
+		if (passivesCompressed) {
 			try {
-				selectedNodes = JSON.parse(savedSelectedNodes);
+				const decompressed = LZString.decompressFromEncodedURIComponent(passivesCompressed);
+				selectedNodes = decompressed ? decompressed.split(',') : [];
 			} catch (error) {
-				console.error('Error parsing saved selected nodes:', error);
+				console.error('Error parsing selected nodes from URL:', error);
 			}
 		}
 	}
 
-	// Reactive statement to save selected nodes to localStorage whenever they change
 	$: if (browser) {
-		localStorage.setItem('selectedSkillNodes', JSON.stringify(selectedNodes));
-		localStorage.setItem('selectedAscendancy', selectedAscendancy);
-	}
+		const nodeIdsString = selectedNodes.join(',');
+		const passivesCompressed = LZString.compressToEncodedURIComponent(nodeIdsString);
 
-	// State for filters
-	let highlightKeystones = false;
-	let highlightNotables = false;
-	let highlightSmalls = false;
-	let hideUnidentified = false;
-	let hideUnselected = false;
-	let hideSmall = false;
+		const params = new URLSearchParams(window.location.search);
+		params.set('a', selectedAscendancy);
+		params.set('p', passivesCompressed);
 
-	// Reactive statement for search
-	$: handleSearch(searchTerm);
+		const newUrl = window.location.pathname + '?' + params.toString();
+		window.history.replaceState({}, '', newUrl);
+	}
 
-	// handler for removing selected nodes when our ascendancy changes
-	$: handleAscendancyChange(selectedAscendancy);
+	let prevSelectedAscendancy = selectedAscendancy;
 
-	function handleAscendancyChange(ascendancy: string) {
-		selectedNodes = selectedNodes.filter((nodeId) => nodes[nodeId].class === ascendancy);
+	$: if (selectedAscendancy !== prevSelectedAscendancy) {
```

Comment:
i'm not particularly experienced with Svelte - do we need to break the ascendency change handler fn out into imperative code? seems like Svelte was handling comparing current/prev values for us prior to this.

Reply:
This behavior is intentional. The previous implementation would wipe out all selected nodes when changing the Ascendancy (we already have a button for that). With this approach, only the Ascendancy related nodes are unselected.

Reply:
rather than blocking getting the feature in, it's probably safe to just send it as is and someone can do a followup PR to extract logic/organise things in a uniform manner, if it's even deemed worth it (given the scope of the project)